cli/pkg/printer/resources.go

```diff
@@ -540,3 +543,106 @@ type billingIssue struct {
+	case runtime.ResourceKindReport:
+		spec, specErr = protojson.MarshalOptions{Indent: " "}.Marshal(r.GetReport().Spec)
+		state, stateErr = protojson.MarshalOptions{Indent: " "}.Marshal(r.GetReport().State)
+	case runtime.ResourceKindAlert:
+		spec, specErr = protojson.MarshalOptions{Indent: " "}.Marshal(r.GetAlert().Spec)
+		state, stateErr = protojson.MarshalOptions{Indent: " "}.Marshal(r.GetAlert().State)
+	case runtime.ResourceKindPullTrigger:
+		spec, specErr = protojson.MarshalOptions{Indent: " "}.Marshal(r.GetPullTrigger().Spec)
+		state, stateErr = protojson.MarshalOptions{Indent: " "}.Marshal(r.GetPullTrigger().State)
+	case runtime.ResourceKindRefreshTrigger:
+		spec, specErr = protojson.MarshalOptions{Indent: " "}.Marshal(r.GetRefreshTrigger().Spec)
+		state, stateErr = protojson.MarshalOptions{Indent: " "}.Marshal(r.GetRefreshTrigger().State)
+	case runtime.ResourceKindTheme:
+		spec, specErr = protojson.MarshalOptions{Indent: " "}.Marshal(r.GetTheme().Spec)
+		state, stateErr = protojson.MarshalOptions{Indent: " "}.Marshal(r.GetTheme().State)
+	case runtime.ResourceKindComponent:
+		spec, specErr = protojson.MarshalOptions{Indent: " "}.Marshal(r.GetComponent().Spec)
+		state, stateErr = protojson.MarshalOptions{Indent: " "}.Marshal(r.GetComponent().State)
+	case runtime.ResourceKindCanvas:
+		spec, specErr = protojson.MarshalOptions{Indent: " "}.Marshal(r.GetCanvas().Spec)
+		state, stateErr = protojson.MarshalOptions{Indent: " "}.Marshal(r.GetCanvas().State)
+	case runtime.ResourceKindAPI:
+		spec, specErr = protojson.MarshalOptions{Indent: " "}.Marshal(r.GetApi().Spec)
+		state, stateErr = protojson.MarshalOptions{Indent: " "}.Marshal(r.GetApi().State)
+	case runtime.ResourceKindConnector:
+		spec, specErr = protojson.MarshalOptions{Indent: " "}.Marshal(r.GetConnector().Spec)
+		state, stateErr = protojson.MarshalOptions{Indent: " "}.Marshal(r.GetConnector().State)
+	default:
+		specErr = fmt.Errorf("unknown resource kind %q", r.Meta.Name.Kind)
+		stateErr = fmt.Errorf("unknown resource kind %q", r.Meta.Name.Kind)
+	}
+
+	if specErr != nil {
+		spec = []byte(specErr.Error())
+	}
+	if stateErr != nil {
+		state = []byte(stateErr.Error())
+	}
+
+	return &resource{
+		Org:          org,
+		Project:      project,
+		ResourceType: runtime.PrettifyResourceKind(r.Meta.Name.Kind),
+		ResourceName: r.Meta.Name.Name,
+		Meta:         meta,
+		Spec:         spec,
+		State:        state,
+	}
+}
+
+type resource struct {
+	Org          string          `header:"org" json:"org"`
+	Project      string          `header:"project" json:"project"`
+	ResourceType string          `header:"resource_type" json:"resource_type"`
+	ResourceName string          `header:"resource_name" json:"resource_name"`
+	Meta         json.RawMessage `header:"meta" json:"meta"`
+	Spec         json.RawMessage `header:"spec" json:"spec"`
+	State        json.RawMessage `header:"state" json:"state"`
+}
```

Comment:
CLI formatted tables don't really make sense here since meta/spec/state are so big. IMO can just remove and only have JSON output from the command itself (like in the first PR).